test/csuite/random_directio/main.c

```diff
@@ -0,0 +1,1127 @@
+ * written by a thread.  T represents the thread id reduced to a single
+ * hex digit. LARGE_STRING is a portion of a large string that includes
+ * the thread id and a lot of spaces, over and over (see the large_buf
+ * function).  When forming the key, the large string is truncated so
+ * that the key is effectively padded to the right length.
+ *
+ * The key space for the main table is designed to be interleaved tightly
+ * among all the threads.  The matching values in the main table are the
+ * same, except with the xxxx string reversed.  So the keys and values
+ * are the same size.
+ *
+ * There is also a reverse table where the keys/values are swapped.
+
+ */
+
+#include "test_util.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <signal.h>
+
+static char home[1024];			/* Program working dir */
+
+/*
+ * These two names for the URI and file system must be maintained in tandem.
+ */
+static const char * const uri_main = "table:main";
+static const char * const fs_main = "main.wt";
+
+static const char * const uri_rev = "table:rev";
+
+/*
+ * The number of threads cannot be more than 16, we are using a hex digit
+ * to encode this in the key.
+ */
+#define	MAX_TH			16
+#define	MIN_TH			5
+
+#define	MAX_TIME		40
+#define	MIN_TIME		10
+
+#define	LARGE_WRITE_SIZE	(128*1024)
+#define	MIN_DATA_SIZE    	30
+#define	DEFAULT_DATA_SIZE	50
+
+#define	DEFAULT_CYCLES		5
+#define	DEFAULT_INTERVAL	3
+
+#define	KEY_SEP			"_"		/* Must be one char string */
+
+#define	ENV_CONFIG							\
+    "create,log=(file_max=10M,enabled),"				\
+    "transaction_sync=(enabled,method=%s)"
+#define	ENV_CONFIG_REC "log=(recover=on)"
+
+/* 64 spaces */
+#define	SPACES								\
+	"                                                                "
+
+/*
+ * Set higher to be less stressful for schema operations.
+ */
+#define	SCHEMA_OP_FREQUENCY	100
+#define	SCHEMA_OP(id, offset)						\
+	(((offset) == 0 || (id) > (offset)) &&				\
+	    (((id) - (offset)) % SCHEMA_OP_FREQUENCY < 10 ||		\
+	    ((id) - (offset)) % SCHEMA_OP_FREQUENCY == 0))
+
+extern int __wt_optind;
+extern char *__wt_optarg;
+
+static void handler(int);
+static void usage(void)
+    WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+static void
+usage(void)
+{
+	fprintf(stderr, "usage: %s [options]\n", progname);
+	fprintf(stderr, "options:\n"
+	    "  -d data_size   \t"
+	    "approximate size of keys and values [1000]\n"
+	    "  -h home        \t"
+	    "WiredTiger home directory [WT_TEST.directio]\n"
+	    "  -i interval    \t"
+	    "interval timeout between copy/recover cycles [3]\n"
+	    "  -m method      \t"
+	    "sync method: fsync, dsync, none [none]\n"
+	    "  -n num_cycles  \t"
+	    "number of copy/recover cycles [5]\n"
+	    "  -p             \t"
+	    "populate only [false]\n"
+	    "  -S             \t"
+	    "schema operations on [false]\n"
+	    "  -T num_threads \t"
+	    "number of threads in writer[ 4]\n"
```

Comment:
The execution of the test with default settings, creates random number of writer threads. So why are we saying that default number of writer threads is 4 in the usage?

Reply:
Right!  Set defaults for -T and -t to [random] in the usage text, to match the code.